crystalfontz/cli.py

```diff
@@ -451,80 +559,60 @@ def main(
     """
+    pass
 
-    def decorator(fn: AsyncCommand) -> WrappedAsyncCommand:
-        @click.pass_context
-        @functools.wraps(fn)
-        def wrapped(ctx: click.Context, *args, **kwargs) -> None:
-            port: str = ctx.obj.port
-            model = ctx.obj.model
-            hardware_rev = ctx.obj.hardware_rev
-            firmware_rev = ctx.obj.firmware_rev
-            detect = ctx.obj.detect
-            output = ctx.obj.output
-            timeout = ctx.obj.timeout
-            retry_times = ctx.obj.retry_times
-            baud_rate: BaudRate = ctx.obj.baud_rate
-
-            report_handler = report_handler_cls()
-
-            # Set the output mode on the report handler
-            if isinstance(report_handler, CliReportHandler):
-                report_handler.mode = output
-
-            # Set the output mode for echo
-            echo.mode = output
 
-            async def main() -> None:
-                try:
-                    client: Client = await create_connection(
-                        port,
-                        model=model,
-                        hardware_rev=hardware_rev,
-                        firmware_rev=firmware_rev,
-                        report_handler=report_handler,
-                        timeout=timeout,
-                        retry_times=retry_times,
-                        baud_rate=baud_rate,
-                    )
-                except SerialException as exc:
-                    click.echo(exc)
-                    sys.exit(1)
+@config.command()
+@click.argument("name")
+@click.pass_obj
+def get(obj: Obj, name: str) -> None:
+    """
+    Get a parameter from the configuration file.
+    """
 
-                # If enabled, detect the device and update it accordingly
-                if detect:
-                    await client.detect_device()
+    echo(obj.config.get(name))
 
-                # Giddyup!
-                await fn(client, *args, **kwargs)
 
-                # Close the client if we're done
-                if not run_forever:
-                    client.close()
+@config.command()
+@click.argument("name")
+@click.argument("value")
+@click.pass_obj
+def set(obj: Obj, name: str, value: str) -> None:
+    """
+    Set a parameter in the configuration file.
+    """
+    obj.config.set(name, value)
+    obj.config.to_file()
 
-                # Await the client closing and surface any exceptions
-                await client.closed
 
-            try:
-                asyncio.run(main())
-            except KeyboardInterrupt:
-                pass
+OTHER_BAUD_RATE: Dict[BaudRate, BaudRate] = {
+    SLOW_BAUD_RATE: FAST_BAUD_RATE,
+    FAST_BAUD_RATE: SLOW_BAUD_RATE,
+}
 
-        return wrapped
 
-    return decorator
+@config.command()
+@pass_client()
+@click.pass_obj
+async def detect(obj: Obj, client: Client) -> None:
+    """
+    Detect device settings and set them in the configuration file.
+    """
+    try:
+        await client.detect_device()
```

Comment:
This may not be good enough. Better would probably be to send a ping with a long message, and assert the pong.